File: java/demo/src/asyneMemManager/clientDemo/model/TestEntity.java
package asyneMemManager.clientDemo.model;

import java.util.Random;

import asyncMemManager.common.di.AsyncMemSerializer;

public class TestEntity {
	private String stringProperty;
	private int[] largeProperty;
	private static final int largePropertySize = 10000000;
	
	public static TestEntity initLargeObject() {
		TestEntity e = new TestEntity();
		e.stringProperty = String.format("this is test string %d", new Random().nextInt());
		e.largeProperty = new int[largePropertySize];
		e.largeProperty[0] = new Random().nextInt();
		return e;
	}
	
	public String getSomeText() {
		return this.stringProperty + this.largeProperty[0];
	}
	
	public static class TestEntityAsyncMemSerializer implements AsyncMemSerializer<TestEntity>
	{
		public static final TestEntityAsyncMemSerializer Instance = new TestEntityAsyncMemSerializer();
		private TestEntityAsyncMemSerializer()
		{			
		}
		
		@Override
		public String serialize(TestEntity object) {
			if (object.largeProperty != null)
			{
				return object.stringProperty;
			}else {
				return "" + object.largeProperty + "##" + object.stringProperty;
			}
		}

		@Override
		public TestEntity deserialize(String data) {
			TestEntity e = new TestEntity();
			int indexOfSplitter = data.indexOf("##");
			if (indexOfSplitter>=0)
			{
				e.largeProperty=new int[largePropertySize];
				e.largeProperty[0] = Integer.parseInt(data.substring(0, indexOfSplitter));
				e.stringProperty = data.substring(indexOfSplitter+2);
			}else {
				e.stringProperty = data;
			}
			return e;
		}

		@Override
		public long estimateObjectSize(TestEntity object) {		
			return largePropertySize + 20;
		}
	}
}

File: .net/Demo/TestEntity.cs
namespace DemoApp.model
{
    using System;
    using AsyncMemManager.Client.DI;

    public class TestEntity {
        private string stringProperty;
        private int[] largeProperty;
        public const int LARGE_PROPERTY_SIZE = 10000;
        
        public static TestEntity InitLargeObject() {
            TestEntity e = new TestEntity();
            e.stringProperty = string.Format("this is test string {0}", new Random().Next());
            e.largeProperty = new int[LARGE_PROPERTY_SIZE];
            e.largeProperty[0] = new Random().Next();
            return e;
        }
        
        public string GetSomeText() {
            if (string.IsNullOrEmpty(stringProperty) || this.largeProperty[0] == 0)
            {
                throw new InvalidOperationException("Invalid Object State");
            }
            
            return this.stringProperty + " " + this.largeProperty[0];
        }
        
        public class TestEntityAsyncMemSerializer : IAsyncMemSerializer<TestEntity>
        {
            public static readonly TestEntityAsyncMemSerializer Instance = new TestEntityAsyncMemSerializer();
            private TestEntityAsyncMemSerializer()
            {			
                
            }
            
            public string Serialize(TestEntity obj) {
                if (obj.largeProperty == null)
                {
                    return obj.stringProperty;
                }else {
                    return "" + obj.largeProperty[0] + "##" + obj.stringProperty;
                }
            }

            public TestEntity Deserialize(string data) {
                TestEntity e = new TestEntity();
                int indexOfSplitter = data.IndexOf("##");
                if (indexOfSplitter >= 0)
                {
                    e.largeProperty=new int[LARGE_PROPERTY_SIZE];
                    e.largeProperty[0] = int.Parse(data.Substring(0, indexOfSplitter));
                    e.stringProperty = data.Substring(indexOfSplitter+2);
                }else {
                    e.stringProperty = data;
                }
                return e;
            }

            public long EstimateObjectSize(TestEntity obj) {		
                return LARGE_PROPERTY_SIZE + 20;
            }
        }
    }    
}
